Add style guide (#270)
smartdevicelink/sdl_app_library_guides: STYLE_GUIDE.md

# Style Guide

## Formatting

> * Remember that you can't nest platform tags.

Reviewer: Fenced code blocks should have a language identifier to enable syntax highlighting. (e.g. ```java)

### Whitespace
* Use spaces, not tabs, 4 spaces per indentation.
* Use one space after a period, not two.

> * One line between paragraphs, paragraphs and code, or paragraphs and a header (###) following it.

Reviewer: ```suggestion
* Add one newline between two paragraphs, a paragraph and code snippet, or a paragraph followed by a header (i.e. any line of text starting with the `#` sign).
```

> * No extra lines between a header (###) and the paragraph following it.

Reviewer: ```suggestion
* Do not add a newline between a paragraph and its header.
```

## Grammar

> * In text, use `SDL`, not `sdl` or `Sdl`.

Reviewer: ```suggestion
* When referring to "SmartDeviceLink" in the text, use _SDL_, not _sdl_ or _Sdl_.
```
